sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -12,19 +12,161 @@
 #include <string>
 
 namespace Azure { namespace Core {
+
+  /**
+   * @brief An exception that gets thrown when @DateTime error occurs.
+   */
+  class DateTimeException : public std::runtime_error {
```

Comment:
For recoverable errors, we should be inheriting from std::exception.
For non-recoverable (precondition-like) errors, we should be inheriting from std::runtime_error.
We should double check this for all exception types we define.